src/platform/testing/e2e/cypress/support/commands/keyboardNavigation.js

```diff
@@ -286,13 +320,52 @@ Cypress.Commands.add('typeInFullName', (fieldName, data) => {
  * @param {typeInDate} callbackFunction - The callback that handles making the month & day selections and typing in the year
  */
 Cypress.Commands.add('typeInDate', (fieldName, dateString) => {
-  // Remove leading zeros
-  const date = dateString.split('-').map(v => parseInt(v, 10).toString());
-  const name = removeLeadingHash(fieldName);
-  cy.tabToElement(`#${name}Month`);
-  cy.chooseSelectOptionUsingValue(date[1]);
-  cy.tabToElement(`#${name}Day`);
-  cy.chooseSelectOptionUsingValue(date[2]);
-  cy.tabToElement(`input[name="${name}Year"]`);
-  cy.typeInFocused(date[0]);
+  if (dateString) {
+    const [year, month, day] = dateString
+      .split('-')
+      .map(v => parseInt(v, 10).toString());
+    const name = removeLeadingHash(fieldName);
+    cy.selectDropdownFromData(`[name="${name}Month"]`, month);
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(day)) {
```

Comment:
This handles the `monthYearOnly` variant by skipping the day input if the value is `XX`.